src/config/keys.ts

```diff
@@ -12,73 +12,147 @@ import { AggregateActionsArgs } from '../utils/action';
+      children: (types?: UnionOfConst<typeof ItemType>[]) =>
+        [...allChildren, types] as const,
+
+      // todo: add page and filtering options
+      paginatedChildren: [...allChildren, 'paginated'] as const,
+
+      // descendants
+      descendants: [...singleBaseKey, 'descendants'] as const,
+
+      // parents
+      parents: [...singleBaseKey, 'parents'] as const,
+
+      // categories
+      categories: [...singleBaseKey, 'categories'] as const,
+
+      // item login
+      itemLoginSchema: {
+        content: [...itemLoginSchema, 'content'] as const,
+        type: [...itemLoginSchema, 'type'] as const,
+      },
+
+      // thumbnails
+      allThumbnails,
+      thumbnail: (options: { size?: string; replyUrl?: boolean }) =>
+        [
+          ...allThumbnails,
+          options.size ?? DEFAULT_THUMBNAIL_SIZE,
+          options.replyUrl ? 'url' : 'blob',
+        ] as const,
+
+      // etherpad
+      etherpad: [...singleBaseKey, 'etherpad'] as const,
+
+      // geolocation
+      geolocation: [...singleBaseKey, 'geolocation'] as const,
+
+      // short links
+      shortLinks: [...singleBaseKey, 'shortLink'] as const,
+
+      // likes
+      likes: [...singleBaseKey, 'likes'] as const,
+
+      // published info
+      publishedInformation: [...singleBaseKey, 'publishedInformation'] as const,
+    };
+  },
+
+  // many items
+  allMany: () => [...itemKeys.all, 'many'] as const,
+  many: (ids: UUID[]) => {
+    const manyBaseKey = [...itemKeys.allMany(), ids];
+    return {
+      content: [...manyBaseKey, 'content'],
+      children: [...manyBaseKey, 'children'],
+      // published info
+      publishedInformation: [...manyBaseKey, 'publishedInformation'],
+    };
+  },
+
+  // accessible items
+  allAccessible: () => [...itemKeys.all, 'accessible'] as const,
+  accessiblePage: (params: ItemSearchParams, pagination: PaginationParams) =>
+    [...itemKeys.allAccessible(), params, pagination] as const,
+
+  // shared items
+  shared: () => [...itemKeys.all, 'shared'] as const,
 };
-export const SHARED_ITEMS_KEY = ['shared'];
-export const CURRENT_MEMBER_KEY = ['currentMember'];
+
+export const memberKeys = {
+  all: ['members'] as const,
+  single: (id?: UUID) => {
+    const singleBaseKey = [...memberKeys.all, id] as const;
+    const allAvatar = [...singleBaseKey, 'avatar'] as const;
+    return {
+      content: [...singleBaseKey] as const,
+      allAvatar,
+      avatar: (options: { size?: string; replyUrl?: boolean }) =>
+        [...allAvatar, { size: DEFAULT_THUMBNAIL_SIZE, ...options }] as const,
+      profile: [...singleBaseKey, 'profile'] as const,
+      likedItems: [...singleBaseKey, 'likedItems'] as const,
+    };
+  },
+  many: (ids?: UUID[]) => [...memberKeys.all, ids] as const,
+  current: () => {
+    const currentBaseKey = [...memberKeys.all, 'current'] as const;
+    return {
+      content: [...currentBaseKey] as const,
+      // should be in items ?
```

Comment:
I think, because the liked items depend on the member, it is fine to keep this key here. Also, this key is available into the `single` property. In this one, you have to give the id of the member and not the id of the item, so for me, it is more the responsibility of the `memberKeys` to manage the liked items key instead of the item.

The boundary is thin, so I would be interested to have your opinions.